### ISSUE TITLE
Add metrics to measure IO efficiency

### ISSUE THREAD

**Issue by aweisberg** [prestodb/presto#16585]
We want to be able to loosely evaluate the IO efficiency of any query and need to come up with some metrics that could help with that.

There are two metrics I have come up with:

**wasted_projection_bytes**
Bytes read during query execution from a datasource (or by a datasource) that are discarded as part of projecting out just the columns we want. This would be an additional statistic in the query completed event.

This is intended to determine if a query could have been more efficient if it was able to leverage a materialized view, better column ordering, or some other layout change to scan less data.

**wasted_filter_bytes**
Bytes read from a datasource (including projected out columns) that are then filtered out by a predicate. For simplicity lets say this only applies to what is pushed all the way down to scan.

This is intended to determine if a query could have been more efficient if it was able to leverage a materialized view, secondary indexes, z-order partitioning, or some other scheme to implement the predicate in a more IO efficient way.

**Comment by yuanzhanhku** [comment]
FYI, we have a new metric tracking framework now where adding a metric will be a  one-line change. And the framework automatically aggregates it at stage level and query level and displays it in the Coordinator UI, Lookup UI and stores it in scuba.
Here are the instructions on how to add new metrics using the new tracking framework: 
https://docs.google.com/document/d/1xkehuyVC50iYKbzV__JiDYYoGmXLlt_fK9HqSkAwJZ8/edit#heading=h.ewd218agafso

**Comment by nmahadevuni** [comment]
> FYI, we have a new metric tracking framework now where adding a metric will be a one-line change. And the framework automatically aggregates it at stage level and query level and displays it in the Coordinator UI, Lookup UI and stores it in scuba.
> Here are the instructions on how to add new metrics using the new tracking framework:
> https://docs.google.com/document/d/1xkehuyVC50iYKbzV__JiDYYoGmXLlt_fK9HqSkAwJZ8/edit#heading=h.ewd218agafso

@yuanzhanhku Is this not a public document?

**Comment by yuanzhanhku** [comment]
> > FYI, we have a new metric tracking framework now where adding a metric will be a one-line change. And the framework automatically aggregates it at stage level and query level and displays it in the Coordinator UI, Lookup UI and stores it in scuba.
> > Here are the instructions on how to add new metrics using the new tracking framework:
> > https://docs.google.com/document/d/1xkehuyVC50iYKbzV__JiDYYoGmXLlt_fK9HqSkAwJZ8/edit#heading=h.ewd218agafso
> 
> @yuanzhanhku Is this not a public document?

Right, this is currently an internal doc. @aweisberg also raised this visibility issue to me. I shared the doc with Jinlin directly. I will make a public document for this topic soon.
